sphinxgallery/gen_rst.py

```diff
@@ -118,133 +78,161 @@ def flush(self):
-
-
-def extract_line_count(filename, target_dir):
-    """Extract the line count of a file"""
-    example_file = os.path.join(target_dir, filename)
-    lines = open(example_file).readlines()
-    start_row = 0
-    if lines and lines[0].startswith('#!'):
-        lines.pop(0)
-        start_row = 1
-    line_iterator = iter(lines)
-    tokens = tokenize.generate_tokens(lambda: next(line_iterator))
-    check_docstring = True
-    erow_docstring = 0
-    for tok_type, _, _, (erow, _), _ in tokens:
-        tok_type = token.tok_name[tok_type]
-        if tok_type in ('NEWLINE', 'COMMENT', 'NL', 'INDENT', 'DEDENT'):
+        # because we only allow for one single docstring
+        elif line.startswith('"""') and has_header_string == 1:
+            has_header_string += 1
+            block += line
+            blocks.append(('text', block))
+            block = ''
             continue
-        elif (tok_type == 'STRING') and check_docstring:
-            erow_docstring = erow
-            check_docstring = False
-    return erow_docstring+1+start_row, erow+1+start_row
-
-
-def line_count_sort(file_list, target_dir):
-    """Sort the list of examples by line-count"""
-    new_list = [x for x in file_list if x.endswith('.py')]
-    unsorted = np.zeros(shape=(len(new_list), 2))
-    unsorted = unsorted.astype(np.object)
-    for count, exmpl in enumerate(new_list):
-        docstr_lines, total_lines = extract_line_count(exmpl, target_dir)
-        unsorted[count][1] = total_lines - docstr_lines
-        unsorted[count][0] = exmpl
-    index = np.lexsort((unsorted[:, 0].astype(np.str),
-                        unsorted[:, 1].astype(np.float)))
-    if not len(unsorted):
-        return []
-    return np.array(unsorted[index][:, 0]).tolist()
-
-
-def generate_dir_rst(src_dir, target_dir, gallery_conf,
-                     plot_gallery, seen_backrefs):
-    """Generate the rst file for an example directory"""
-    if not os.path.exists(os.path.join(src_dir, 'README.txt')):
-        print(80 * '_')
-        print('Example directory %s does not have a README.txt file' %
-              src_dir)
-        print('Skipping this directory')
-        print(80 * '_')
-        return ""  # because string is an expected return type
 
-    fhindex = open(os.path.join(src_dir, 'README.txt')).read()
-    if not os.path.exists(target_dir):
-        os.makedirs(target_dir)
-    sorted_listdir = line_count_sort(os.listdir(src_dir),
-                                     src_dir)
-    for fname in sorted_listdir:
-        if fname.endswith('py'):
-            generate_file_rst(fname, target_dir, src_dir, plot_gallery)
-            new_fname = os.path.join(src_dir, fname)
-            _, snippet, _ = extract_docstring(new_fname, True)
-            write_backreferences(seen_backrefs, gallery_conf,
-                               target_dir, fname, snippet)
+        # comment blocks
+        if line.startswith('#') and not continue_text:
```

Comment:
OK fair enough.

FWIW unclean is a bit harsh, strings for multi-line comments was mentioned by Guido Van Rossum [here](https://twitter.com/gvanrossum/status/112670605505077248).